## Task 11 — Revenue by Country

### Business Question

Which countries generate the highest average monthly revenue per customer?

### Analysis

Group customers by country, calculate total monthly revenue and the number of customers for each country, then calculate average monthly revenue per customer and sort the countries in descending order by this metric.

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
total_revenue_by_country = starlink.groupby("country").agg(
    total_revenue = ("monthly_fee","sum"),
    users_amount = ("unique_id","count")
)
total_revenue_by_country["average_revenue_per_customer"] = total_revenue_by_country["total_revenue"] / total_revenue_by_country["users_amount"]
total_revenue_by_country_sorted = total_revenue_by_country.sort_values(
    "average_revenue_per_customer",
    ascending = False
)
print(total_revenue_by_country_sorted)

                total_revenue  users_amount  average_revenue_per_customer
country                                                                  
Japan                117710.0           973                    120.976362
Canada               119630.0           997                    119.989970
Spain                122100.0          1019                    119.823356
United States        114250.0           954                    119.758910
Australia            122360.0          1030                    118.796117
France               117780.0           996                    118.253012
Ukraine              121760.0          1032                    117.984496
Brazil               115680.0           996                    116.144578
United Kingdom       113450.0           977                    116.120778
Germany              118550.0          1027                    115.433301


### Result

Japan has the highest average monthly revenue per customer at 120.98, followed by Canada at 119.99 and Spain at 119.82. Germany has the lowest average revenue per customer at 115.43.

### Business Insight

Average revenue per customer varies only moderately across countries, despite differences in total revenue. Japan generates the highest revenue per customer, while Australia generates the highest total revenue due to its larger customer base. This indicates that customer volume has a stronger impact on total revenue than small differences in average revenue per customer.